### PR TITLE
Tp python api work

### PR DESCRIPTION
**Pull request by tylerpuleo**

### CODE REVIEW

File: Python/class/API.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import requests,json,time,datetime
+
+
+class Isense:
+
+    def __init__(self,projectID,contributorKey,contributorName):
+        self.projectID = projectID
+        self.contributorKey = contributorKey
+        self.contributorName = contributorName
+        self.baseUrl = 'http://rsense-dev.cs.uml.edu/api/v1/';
```
Comment (asalani93) [inline Python/class/API.py]: baseUrl is never modified - each instance of Isense API doesn't need it's own.  Granted, there will probably ever only be one instance, but it's more of a best practices kind of thing.

File: Python/class/API.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+    def projectGetRequest(self):
+        
+        urlProject = self.baseUrl+ 'projects/' + self.projectID+'?recur=true';
```
Comment (asalani93) [inline Python/class/API.py]: consistent whitespace between operators

File: Python/class/API.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    def getDatasetId(self,datasetName,parsedResponseProject):
+
+        for i in range(0,parsedResponseProject.json()['dataSetCount'] - 1):
+
+            if parsedResponseProject.json()['dataSets'][i]['name'] == datasetName:
+                datasetLocation = i;
+
+                datasetID = parsedResponseProject.json()['dataSets'][i]['id'];
+
+        if datasetID == None:
```
Comment (asalani93) [inline Python/class/API.py]: best practice would be to declare datasetID outside of the for loop

File: Python/class/API.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            if parsedResponseProject.json()['dataSets'][i]['name'] == datasetName:
+                datasetLocation = i
+                datasetID = parsedResponseProject.json()['dataSets'][i]['id']
+                return datasetLocation
+
+        return 'Dataset not found'
+
+    def getDatasetId(self,datasetName,parsedResponseProject):
+
+        for i in range(0,parsedResponseProject.json()['dataSetCount'] - 1):
```
Comment (asalani93) [inline Python/class/API.py]: why is this dataSetCount - 1, and the above function uses just dataSetCount?

File: Python/class/API.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+    def getFieldID(self,fieldName,parsedResponseProject):
+
+        for i in range(0,parsedResponseProject.json()['fieldCount']):
+
+            if parsedResponseProject.json()['fields'][i]['name'] == fieldName:
+
+                fieldID = parsedResponseProject.json()['fields'][i]['id']
+                fieldID = str(fieldID)
+                return fieldID;
```
Comment (asalani93) [inline Python/class/API.py]: same as above

File: Python/class/API.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    def getDatasetLocation(self,datasetName,parsedResponseProject):
+
+        for i in range(0,parsedResponseProject.json()['dataSetCount']):
+
+            if parsedResponseProject.json()['dataSets'][i]['name'] == datasetName:
+                datasetLocation = i
+                datasetID = parsedResponseProject.json()['dataSets'][i]['id']
+                return datasetLocation
+
+        return 'Dataset not found'
```
Comment (asalani93) [inline Python/class/API.py]: Return None or throw an exception, don't return a string.

File: Python/class/API.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+      
+    def getDatasetFieldData(self,datasetName,fieldName):
+
+        values = []
+
+        parsedResponseProject = self.projectGetRequest()
+
+        datasetLocation = self.getDatasetLocation(datasetName,parsedResponseProject)
+
+        fieldID = self.getFieldID(fieldName,parsedResponseProject)
```
Comment (asalani93) [inline Python/class/API.py]: all of these can return values that are not field IDs, should error check that

File: Python/class/API.py [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+        return values
+
+    def postDataset(self,fieldName,datasetName,fieldData):
+        
+        timestamp = time.time()
+        timestamp_reformatted = datetime.datetime.fromtimestamp(timestamp).strftime('%Y-%m-%d %H:%M:%S')
+
+        parsedResponseProject = self.projectGetRequest()
+        fieldID = self.getFieldID(fieldName,parsedResponseProject)
```
Comment (asalani93) [inline Python/class/API.py]: same as above re: error checking

File: Python/class/API.py [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+            'data':
+            {
+                fieldID: fieldData
+            }
+        }
+        headers = {'content-type': 'application/json'}
+
+        r = requests.post(url, data=json.dumps(payload), headers=headers)
+
+        print "\nPost Complete"
```
Comment (asalani93) [inline Python/class/API.py]: never print to stdout from a library

File: Python/class/API.py [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+            'contribution_key': self.contributorKey,                    
+            'contributor_name': self.contributorName,
+            'data': dataForPost    
+        }
+
+        headers = {'content-type': 'application/json'}
+
+        r = requests.post(url, data=json.dumps(payload), headers=headers)
+
+        print "Post Successful"
```
Comment (asalani93) [inline Python/class/API.py]: never ever print to stdout

File: Python/class/PostExample.py [hunk 11]
```diff
@@ -1,7 +1,11 @@
...
-print data
+print x.getDatasetFieldData('MLB Team Statistics 2013','Runs')
 
-IsenseModule.postDataset('106','Key','Calculated Runs','TEST','TYLER',data)
\ No newline at end of file
+x.postDataset('Runs','MLB Team Statistics 2013',[5])
+
+x.postDatasetHorizontal(['Runs','Hits'],'TESTETSTES',[[5],[2]])
+
+x.appendToDataset('TESTETSTES 2015-03-18 17:26:40',['Runs','Hits'],[[5],[2]])
```
Comment (asalani93) [inline Python/class/PostExample.py]: will this timestamp always be valid?  is this the actual name of the dataset?  couldn't it be better?

File: Python/class/PostExample.py [hunk 12]
```diff
@@ -1,7 +1,11 @@
...
+x = API.Isense('106','key','TYLER')
 
-print data
+print x.getDatasetFieldData('MLB Team Statistics 2013','Runs')
 
-IsenseModule.postDataset('106','Key','Calculated Runs','TEST','TYLER',data)
\ No newline at end of file
+x.postDataset('Runs','MLB Team Statistics 2013',[5])
+
+x.postDatasetHorizontal(['Runs','Hits'],'TESTETSTES',[[5],[2]])
```
Comment (asalani93) [inline Python/class/PostExample.py]: do not post data to the testestest data set.  use a real name

**Comment by tylerpuleo** [comment]
@asalani93

File: Python/class/API.py (hunk 7 above)
Comment (asalani93) [inline Python/class/API.py]: getDatasetLocation can still return None

**Pull request merged**